clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -708,6 +708,8 @@ public void handleResponse(AbstractResponse response) {
                     return;
                 } else if (error == Errors.TOPIC_AUTHORIZATION_FAILED) {
                     unauthorizedTopics.add(topicPartition.topic());
+                } else if (error == Errors.OPERATION_NOT_ATTEMPTED) {
+                    log.info("{}Did not attempt to add partition {} to transaction because other partitions in the batch had errors.", logPrefix, topicPartition);
```

Comment:
I agree that this should be `debug`. In theory, the partitions that have caused `OPERATION_NOT_ATTEMPTED` would have the more useful information. @hachikuji, are you simply suggesting that `hasPartitionErrors` should be set to `true` in this path too? It seems like that does what you're basically suggesting already.